packages/wallet/src/redux/protocols/existing-ledger-funding/reducer.ts

```diff
@@ -265,12 +266,42 @@ function ledgerChannelNeedsTopUp(
 
 function craftAppFunding(
   appChannelId: string,
-  allocation: string[],
+  ledgerChannelId: string,
+  startingAllocation: string[],
+  sharedData: SharedData,
 ): { proposedAllocation: string[]; proposedDestination: string[] } {
-  const total = allocation.reduce(addHex);
+  const { allocation: ledgerAllocation, destination: ledgerDestination } = getLatestCommitment(
+    ledgerChannelId,
+    sharedData,
+  );
+
+  const appTotal = startingAllocation.reduce(addHex);
+
+  // If the ledger allocation is greater than the startingAllocation requested
+  // we subtract the startingAllocation from the ledger allocation
+  const updatedLedgerAllocation = ledgerAllocation.map(a => {
+    const index = ledgerAllocation.indexOf(a);
+    const updatedEntry = subHex(a, startingAllocation[index]);
+
+    if (bigNumberify(updatedEntry).lt(0)) {
+      throw new Error(
```

Comment:
Should the callers of `craftAppFunding` catch this error, and move to a failure state?

Reply:
This check was incorrect (as a ledger channel can be underfunded and will be topped up). I've updated the logic here.